pkg/metrics/otelCounter.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+// otelCounter is a wrapper around otel.Counter
+
+type otelCounter struct {
+	counter metric.Int64Counter
+	ctx     context.Context
+	attrs   []metric.AddOption
+}
+
+func (c *otelCounter) Inc(value int64) {
+	c.counter.Add(c.ctx, value, c.attrs...)
+}
+
+func getAttributes(tags map[string]string) []metric.AddOption {
+	var options []metric.AddOption
+	for k, v := range tags {
+		options = append(options, metric.WithAttributes(attribute.String(k, v)))
```

Comment:
this is still very inefficient, you're forcing the instrument to process N options on every call to Inc(). OTEL has the notion of AttributeSet for this scenario. You create a set via `attribute.NewSet` and convert it into a single option via `WithAttributeSet()`

Reply:
one useful thing you may want to do is to add a benchmark for bumping different kinds of metrics and compare your new OTEL-based implementation and Prometheus-based that we currently use.

Reply:
Working on it.

Reply:
I have made the changes, but should I close this PR and move on to creating the telemetry container?